### ISSUE TITLE
APICompat resources are not shipping in the SDK - missing localization

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#29598]
### Describe the bug
I noticed that a number of assemblies in APICompat define resources which should produce localized resource dlls, however these are not included in the SDK layout.

### To Reproduce
Examine SDK folder like C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\tools\net7.0 for any resources.dlls containing localized resources.

Expect: 
Microsoft.DotNet.ApiCompat.Shared
Microsoft.DotNet.ApiCompatibility
Microsoft.DotNet.PackageValidation

Should all have localized resource DLLs since these define string resources.

Actual:
Resources dlls are missing 

This doesn't appear to be a regression, it looks like these were never included.

As part of this work we should consider if we want all of these files to be localized, or if instead we try to consolidate the resources to fewer assemblies.

**Comment by ViktorHofer** [comment]
> Microsoft.DotNet.ApiCompat.Shared

You can remove that one from your list. That's a shared project that doesn't produce an output assembly. Instead, the resource is inlined into the frontends: ApiCompat.Task and ApiCompat.Tool.

**Comment by ViktorHofer** [comment]
Just verified that this doesn't impact the shipping Microsoft.DotNet.ApiCompat.Task.nupkg and Microsoft.DotNet.ApiCompat.Tool.nupkg. That means that the issue is in the SDK integration: https://github.com/dotnet/sdk/blob/baa2d322f9021029b6cf94f2d580b497e024461b/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L57-L60

@marcpopMSFT @dsplaisted can someone help us make sure that the localization files are correctly picked up? It would also be great if we could omit the transitive dependencies and just P2P `Microsoft.DotNet.ApiCompat.Task.csproj` in this project.

**Comment by marcpopMSFT** [comment]
@ViktorHofer sorry, we don't look at issues tagged for other areas normally. For this issue, I'm not quite clear on what the ask is. Are you looking for guidance or something else?

**Comment by ViktorHofer** [comment]
Guidance, yes. APICompat is currently bundled into the SDK via the ProjectReference in `Microsoft.NEt.Build.Tasks.csproj` (see above). What needs to be done to also include the satellite assemblies?

**Comment by ericstj** [comment]
I took a look at this and I see that Microsoft.Build.Tasks sets it's output directory to https://github.com/dotnet/sdk/blob/37c8b6fa7105844de187d4d9891967cf67887cc5/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L7
To place all it's content in the SDK.

I found that the way that APICompat assemblies were referenced led them to bypass RAR, and thus bypass [the related file gathering](https://github.com/dotnet/msbuild/blob/caf06d19326f9e7dcc06d4d4f4b04f1c6ddae5f0/src/Tasks/AssemblyDependency/ReferenceTable.cs#L950) that it does:
https://github.com/dotnet/sdk/blob/37c8b6fa7105844de187d4d9891967cf67887cc5/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L57-L60

If I remove `OutputItemType="ReferenceCopyLocalPaths" ReferenceOutputAssembly="false"` it helps.  At first doing that I see type-conflicts (CS0436 warning as error) because `Microsoft.DotNet.ApiCompat.Task` defines `TaskBase`.  I suppressed this by applying an alias to these to references so that they'd still flow through RAR but not be considered when compiling any of the `Microsoft.NET.Build.Tasks` sources since they wouldn't use that alias.  I also needed to suppress some dependencies we didn't want to include in the sdk.

I feel like we can clean up how all these project references work a bit.  There's a lot of hacking going on in that task project.
